test/parse_performance_test.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('ArgParser.parse()', () {
+    test('is fast', () {
+      var parser = ArgParser()..addFlag('flag');
+
+      var baseSize = 10000;
+      var baseList = List<String>.generate(baseSize, (_) => '--flag');
+
+      var multiplier = 10;
+      var largeList =
+          List<String>.generate(baseSize * multiplier, (_) => '--flag');
+      var baseTime = _time(() => parser.parse(baseList));
+      var largeTime = _time(() => parser.parse(largeList));
```

Comment:
nit: do one of these in `setUp`. This is likely capturing a lot of jit warmup time for the base list parse time.

Reply:
Right now this test doesn't print anything--I edited it locally to get the numbers on the review thread. If I improve the benchmark, should I also make it print timings every time?

Reply:
You could print - but I was more concerned about the multiplier actually being skewed a lot because of the base time being much higher than it should be :)

Reply:
Done--added the print, it seems harmless enough and it's kind of nice to have.

With the warmup the numbers are much closer to 10x, as expected. They're also faster, so I increased the 'base' number so the whole thing takes about a second, to reduce the noise while not annoying people too much.

```
Parsed 200000 elements in 87ms, 2000000 elements in 818ms.
```